File: Level1/Materal.Workflow/Materal.Workflow/StepDatas/Base/StepData.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Materal.Workflow.StepDatas
{
    /// <summary>
    /// 步骤数据
    /// </summary>
    public abstract class StepData : IStepData
    {
        /// <summary>
        /// 节点数据类型名称
        /// </summary>
        public string StepDataType => GetType().Name;
        /// <summary>
        /// 名称
        /// </summary>
        [Required]
        public virtual string Name { get; set; } = string.Empty;
        /// <summary>
        /// 描述
        /// </summary>
        [System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public virtual string? Description { get; set; }
        /// <summary>
        /// 构建数据
        /// </summary>
        [System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public Dictionary<string, object?>? BuildData { get; set; }
    }
}

File: Level1/Materal.Workflow/Materal.Workflow.Web/scripts/StepDatas/Base/StepData.ts
export abstract class StepData {
    /**
     * 节点数据类型
     */
    private StepDataType: string;
    get StepDataTypeName(): string {
        return this.StepDataType;
    }
    /**
     * 唯一标识
     */
    public ID: string;
    /**
     * 名称
     */
    public Name: string;
    /**
     * 描述
     */
    public Description: string | null;
    /**
     * 构建数据
     */
    public BuildData: any = {};
    constructor(stepDataType: string, id: string) {
        this.StepDataType = stepDataType;
        this.ID = id;
    }
    public abstract RemoveChild(childID: string): void;
}
